utils/dataset_utils.py

```diff
@@ -109,7 +109,7 @@ def initialize_cache_hub_repo(cache_path, dataset_cache_dir):
     repo.lfs_track(["*.feather"])
     return repo
 
-def pull_cache_from_hub(cache_path, dataset_cache_dir):
+def pull_cache_from_hub(cache_path, dataset_cache_dir, logs=None):
```

Comment:
This ends up having to check whether logs is None a lot, which makes the code a bit less readable and I don't think it's preferable to simply having logs defined at the start of this file, thusly:
`logs = utils.prepare_logging(__file__)`
Then this function would look something like this:


```
def pull_cache_from_hub(cache_path, dataset_cache_dir):
    """
    This function tries to pull a datasets cache from the huggingface hub if a
    cache for the dataset does not already exist locally. The function expects you
    to have you HUB_CACHE_ORGANIZATION=<the organization you've set up on the hub to store your cache>
    and HF_TOKEN=<your hf token> on separate lines in a file named .env at the root of this repo.
    Args:
        cache_path (string):
            The path to the local dataset cache that you want.
        dataset_cache_dir (string):
            The name of the dataset repo on the huggingface hub.
    """

    hub_cache_organization, hf_token = _load_dotenv_for_cache_on_hub()
    clone_source = pjoin(hub_cache_organization, dataset_cache_dir)

    if isdir(cache_path):
            logs.info("Already a local cache for the dataset, so not pulling from the hub.")
    else:
            # Here, dataset_info.id is of the form: <hub cache organization>/<dataset cache dir>
            hub_cache = [dataset_info.id.split("/")[-1] for dataset_info in list_datasets(author=hub_cache_organization, use_auth_token=hf_token)]
            if dataset_cache_dir in hub_cache:
                Repository(local_dir=cache_path,
                       clone_from=clone_source,
                       repo_type="dataset", use_auth_token=hf_token)
                logs.info("Pulled cache from hub!")  
            else:
                 logs.info("Asking to pull cache from hub but cannot find cached repo on the hub.")
                 logs.warning("Asking to pull cache from hub but cannot find cached repo on the hub! 😭")

```